src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestEventCounter.cs

```diff
@@ -135,29 +135,46 @@ private void Test_Write_Metric(Listener listener)
 
 
                 /*************************************************************************/
+                int num100msecTimerTicks = 0;
                 tests.Add(new SubTest("EventCounter: Log multiple events in multiple periods",
                     delegate ()
                     {
-                        listener.EnableTimer(logger, .1); /* Poll every .1 s */
-                                                          // logs at 0 seconds because of EnableTimer command
-                        Sleep(100);
-                        logger.Request(1);
-                        Sleep(100);
-                        logger.Request(2);
-                        logger.Error();
-                        Sleep(100);
-                        logger.Request(4);
-                        Sleep(100);
-                        logger.Error();
-                        logger.Request(8);
-                        Sleep(100);
-                        logger.Request(16);
-                        Sleep(200);
-                        listener.EnableTimer(logger, 0);
+                        // We have had problems with timer ticks not being called back 100% reliably.  
+                        // However timers really don't have a strong guarentee (only that the happen eventually)
+                        // So what we do is create a timer callback that simply counts the number of callbacks.
+                        // This acts as a marker to show whether the timer callbacks are happening promptly.  
+                        // If we don't get enough of these tick callbacks then we don't require EventCounter to
+                        // be sending periodic callbacks either.   
+                        num100msecTimerTicks = 0;
+                        using (var timer = new System.Threading.Timer(state => num100msecTimerTicks++, null, 100, 100))
+                        {
+                            Console.WriteLine("EventCounter Multi-event: Enabling logger at {0:mm:ss.fff}, thread {1}", DateTime.UtcNow, Thread.CurrentThread.ManagedThreadId);
```

Comment:
Can we please avoid writing to the console in tests? It makes it more noisy and ugly when devs run tests. Perhaps you can remove them or comment them out?

Reply:
> Can we please avoid writing to the console in tests? 

I do see that it can make things verbose, but I think the alternative is worse (which is no information).  These logging messages are there because I needed them to debug failures that did not repo locally.    Sure I could comment them out, but it means I have to comment them back in 'next time'.  

I think a reasonable compromise is to put them in for tests where they have proven their value and are likely to have value in the future (that is things that are fragile with respect to timing or the platform).   (Which, sadly includes this test...).  

It is true, however that I could get away with fewer log messages.  I will see if I can economize.

Reply:
The problem is that 99.999% of the time the messages will not be relevant to the dev watching their console scroll by. Maybe you could put behind `#if DEBUGGING_TRACING_TESTS` or suchlike?
We are trying to have zero console output...

Reply:
Instead of always writing out, could you store the debug data into a string that's then only written out on failure, or included in the exception messages / asserts on failure?

Reply:
I fixed this so that we only create console output on failure.  

Basically the little harness catches exceptions and puts logging messages into the exception.